raster/src/main/scala/geotrellis/raster/io/geotiff/reader/TiffTagsReader.scala

```diff
@@ -408,27 +408,31 @@ object TiffTagsReader {
       }
     }
 
-    def setColorMap(tiffTags: TiffTags, shorts: Array[Int]): TiffTags =
+    def setColorMap(tiffTags: TiffTags, shorts: Array[Int]): TiffTags = {
       if ((tiffTags &|->
         TiffTags._basicTags ^|->
         BasicTags._photometricInterp get) == 3) {
-        val divider = shorts.size / 3
+        // In GDAL world, `divider` ends up being the same as `bitsPerSample`
+        // but theoretically it's valid to have color tables that are smaller
+        val divider = shorts.length / 3
 
         val arr = Array.ofDim[(Short, Short, Short)](divider)
         cfor(0)(_ < divider, _ + 1) { i =>
           arr(i) = (
             shorts(i).toShort,
             shorts(i + divider).toShort,
             shorts(i + 2 * divider).toShort
-          )
+            )
         }
 
         (tiffTags &|->
           TiffTags._basicTags ^|->
           BasicTags._colorMap set arr.toSeq)
-      } else throw new MalformedGeoTiffException(
+      }
+      else throw new MalformedGeoTiffException(
         "Colormap without Photometric Interpetation = 3."
       )
+    }
```

Comment:
Better to leave off the curly brackets if it's a single statement method (the if counts as a single statement)